### PR TITLE
Product filters

### PR DESCRIPTION
**Pull request by arpit01923**
Added filters for product page

https://user-images.githubusercontent.com/86866251/160225582-9e636758-55aa-4e84-85bc-a983dbb5d922.mp4

### CODE REVIEW

File: src/contexts/ProductReducer.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const ProductProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, defaultState);
+
+  return (
+    <ProductContext.Provider value={{ state, dispatch }}>
+      {children}
+    </ProductContext.Provider>
+  );
+};
```
Comment (rohanmathur91) [inline src/contexts/ProductReducer.jsx]: you are naming a file as ProductReducer and inside that you are creating context, try to put context in another file

File: src/pages/product/main/Main.jsx [hunk 2]
```diff
@@ -2,18 +2,70 @@ import { Header } from "./Header";
...
+    return products;
+  }
+  return categories.reduce(
+    (acc, category) => [
+      ...acc,
+      ...array.filter((product) => product.title === category),
+    ],
+    []
+  );
+};
```
Comment (rohanmathur91) [inline src/pages/product/main/Main.jsx]: You can put this in utils folder

**Review by rohanmathur91 — APPROVED** [review]
Added few changes.

File: src/contexts/ProductReducer.jsx (hunk 1 above)
Comment (arpit01923) [reply src/contexts/ProductReducer.jsx]: changed

File: src/pages/product/main/Main.jsx (hunk 2 above)
Comment (arpit01923) [reply src/pages/product/main/Main.jsx]: done

**Review by ishitakeshawani — APPROVED** [review]
- [x]   Clean and readable code.

- [x] UI is good.

**Pull request merged**